### PR TITLE
CAD-508: update traces for Tx throughput benchmarking

### PR DESCRIPTION
**Pull request by deepfire**
1. trace txid lists in the blockfetch protocol's `AddBlock` message
1. establish order among the `ToObjectOrphans`'s instances -- all now sorted alphabetically, keyed by the unqualified name of their outermost type.
1. a dependency for https://github.com/input-output-hk/cardano-benchmarking/pull/2

Checklist
---------
- [ ] This PR contains all the work required to resolve the linked issue.

- [x] The work contained has sufficient documentation to describe what it does and how to do it.

- [ ] The work has sufficient tests and/or testing.

- [x] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [x] I have added the appropriate labels to this PR.

### CODE REVIEW

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs [hunk 1]
```diff
@@ -412,13 +412,16 @@ instance Show peer => Transformable Text IO [TraceLabelPeer peer
   trTransformer _ verb tr = trStructured verb tr
 
 -- transform @BlockFetchDecision@
-instance Show peer => Transformable Text IO (TraceLabelPeer peer
-                                (TraceFetchClientState header)) where
+instance {-# OVERLAPS #-}
```
Comment (CodiePP) [inline cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: why does this type overlap with another? and with which? can they be put together in the file?
what caused you to introduce overlapping types?

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs [hunk 2]
```diff
@@ -450,6 +453,58 @@ instance ( Condense (HeaderHash blk)
   -- user defined formatting of log output
   trTransformer _ verb tr = trStructured verb tr
 
+instance DefinePrivacyAnnotation a => DefinePrivacyAnnotation (TraceLabelPeer peer a)
```
Comment (CodiePP) [inline cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: OK, because of these generic instances?
so what did you gain? I see 78 additions and only 21 deletions in this file.

File: shell.nix [hunk 3]
```diff
@@ -22,9 +22,6 @@ let
        # in theory we should only have the above two packages (or better, they should be auto-detected),
        # but due to source-repository-package declarations being considered as local packages by cabal, we need the following packages as well.
        # cf. https://github.com/haskell/cabal/issues/6249 and https://github.com/haskell/cabal/issues/5444
-       ps.cardano-sl-x509
```
Comment (CodiePP) [inline shell.nix]: these changes are not related to this PR.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs [hunk 4]
```diff
@@ -450,6 +453,58 @@ instance ( Condense (HeaderHash blk)
...
+         , HasHeader blk
+         , HasTxs blk
+         , HasTxId (GenTx blk)
+         )
+ => ToObject (AnyMessage (BlockFetch blk)) where
+  toObject _verb (AnyMessage (MsgBlock blk)) =
+    mkObject
+      [ "kind" .= String "MsgBlock"
+      , "blkid" .= String (pack . condense $ blockHash blk)
+      , "txids" .= Array (V.fromList . fmap presentTx $ extractTxs blk)
```
Comment (CodiePP) [inline cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: the newly introduced dependency "vector" is used only once, exactly here. Is this necessary?
a list of "String"s would also be acceptable. we do not need "Array" here.

File: shell.nix (hunk 3 above)
Comment (deepfire) [reply shell.nix]: @CodiePP, I need these changes when I work on the node, and basically no one else cares.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs (hunk 4 above)
Comment (deepfire) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: @CodiePP, I'm sorry, I don't understand -- there isn't an Aeson type with a `[]` -- only one with a vector.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs (hunk 2 above)
Comment (deepfire) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: I'll try to illustrate the difference using a commit that removes the generic instances.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs (hunk 1 above)
Comment (deepfire) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: That's a bit odd.. I just dropped the overlaps and all is well now.  I clearly remember needing them.

Gone, anyway.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs (hunk 2 above)
Comment (deepfire) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: The overlaps are gone.

File: cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs (hunk 4 above)
Comment (deepfire) [reply cardano-config/src/Cardano/Tracing/ToObjectOrphans.hs]: There isn't an Aeson Value ADT variant with a [] -- only one with a vector -- but we can reuse `toJSON` to avoid the dependency

Good point.

File: shell.nix (hunk 3 above)
Comment (deepfire) [reply shell.nix]: Dropped, as I finally got https://github.com/input-output-hk/cardano-node/pull/614 reviewed (was hanging since Feb 28).

**Comment by deepfire** [comment]
@CodiePP, I think I've already done what we discussed, so to prevent further confusion I went to the town and sorted _all_ instances by the unqualified name of the outermost type.
